notebooks/concurrent_tasks.ipynb

```diff
@@ -0,0 +1,1226 @@
+{
```

Comment:
<p>Is asyncio technically using threading? I thought it was single-threaded execution.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepset-ai/haystack-cookbook/pull/81/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

Reply:
I'm using [`asyncio.to_thread`](https://docs.python.org/3/library/asyncio-task.html#running-in-threads): it creates separate threads. In this case, asyncio is only used as a high-level API.